packages/table/src/interactions/reorderable.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+import { Utils } from "../common/utils";
+import { Draggable, ICoordinateData, IDraggableProps } from "../interactions/draggable";
+import { IRegion, RegionCardinality, Regions } from "../regions";
+
+export interface IReorderableProps {
+    /**
+     * A callback that is called while the user is dragging to reorder.
+     *
+     * @param oldIndex the original index of the element or set of elements
+     * @param newIndex the new index of the element or set of elements
+     * @param length the number of contiguous elements that were moved
+     */
+    onReordering: (oldIndex: number, newIndex: number, length: number) => void;
+
+    /**
+     * A callback that is called when the user is done dragging to reorder.
+     *
+     * @param oldIndex the original index of the element or set of elements
+     * @param newIndex the new index of the element or set of elements
+     * @param length the number of contiguous elements that were moved
+     */
+    onReordered: (oldIndex: number, newIndex: number, length: number) => void;
```

Comment:
Yeah, I couldn't find great precedent for this in the codebase. Slider does `onChange` and `onRelease`, `DragSelectable` now has `onSelection` and `onSelectionEnd` as of a recent commit in this PR—do you think `onReordering` still makes sense if we change `onReordered` to `onReorderEnd`?